### ISSUE TITLE
Upgrade from 4.x to 5.x, Downgrade from 5.x to 4.x

### ISSUE THREAD

**Issue by fstr** [voxpupuli/puppet-nodejs#197]
Just changing the `repo_url_suffix` from `node_4.x` to `node_5.x` and set ensure to `present` or `latest` doesn't do any changes.

Version 4.x will still be installed, since it fulfills the `present` requirement.

A way to force a new version is to set a certain version. 
For an upgrade:
```puppet
class { 'nodejs':
  manage_package_repo => true,
  repo_url_suffix => 'node_5.x',
  nodejs_package_ensure => '5.4.0-1nodesource1~wheezy1' # or latest
}
```

`latest` does not work for a downgrade though, so for a downgrade you have to do something like this:
```puppet
class { 'nodejs':
  manage_package_repo => true,
  repo_url_suffix => 'node_4.x',
  nodejs_package_ensure => '4.2.4-1nodesource1~wheezy1'
}
```

Sadly, the apt.pp file only updates the sources if the nodejs_package_ensure is set to `present`

```
11:  if ($ensure == 'present') {
```

Maybe I got it all wrong and there's another way to upgrade/downgrade major versions. Otherwise I'd write a PR.

So I propose a change

```
11:   if ($ensure != 'absent') {
```

**Comment by igalic** [comment]
:+1: @fstr!
would you like to provide us with a pull request?

**Comment by fstr** [comment]
I was working with the 0.8.0 release from puppetlabs, you should publish a new release there!

But the problem seems to persist/be slightly different for the latest 1.3.0 release.

When I change the `repo_url_suffix` from 4.x to 5.x, the sources.list.d file is updated, but no `apt-get update` is executed. A `notify => Exec['apt_update']` seems to be propagated to be executed AFTER `apt-get install`

With the change I proposed providing a certain version, the install fails because of the missing apt-get update.

The best fix I found so far, is doing a `require => Exec['apt_update']` in install.pp for each package to make sure, that the required versions are available after providing the sources.list.

Example:
```
  # nodejs
  package { $nodejs::nodejs_package_name:
    ensure => $nodejs::nodejs_package_ensure,
    require  => Exec['apt_update']
  }
```

I'll try to provide a PR later.

**Comment by igalic** [comment]
> I was working with the 0.8.0 release from puppetlabs, you should publish a new release there!

@fstr puppetlabs/nodejs has been moved to voxpupuli (on the forge this is the /puppet namespace)
we're working together with puppetlabs to make pmt and the forge do redirects.

**Comment by juniorsysadmin** [comment]
Proposed fix merged in #198
